Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Tests;
+
+use PhpAmqpLib\Connection\AMQPStreamConnection;
+use PhpAmqpLib\Exception\AMQPConnectionClosedException;
+use PhpAmqpLib\Message\AMQPMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\AsyncQueueConsumer;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\PhpAmqpLibDriver;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\QueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\ConsumerInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
+use Smartbox\Integration\FrameworkBundle\Tests\Functional\Drivers\Queue\AbstractQueueDriverTest;
+
+/**
+ * Class PhpAmqpLibDriverTest
+ * @group php-amqp-lib
+ */
+class PhpAmqpLibDriverTest extends AbstractQueueDriverTest
```

Comment:
If I am not mistaken, these tests are functional test between the driver and the queue system. Based on this the consumer wouldn't be needed here

Reply:
Pending to review once the changes of the previous comment has been addressed

Reply:
I did a refactor in this class to get a consumer mock instead of the real class.

Reply:
Refactored.

Reply:
:-1: `$consumer` property is still present in this file.

Plus the `getConsumerName()` function doesnt need to emulate what the real consumer does. You could discard the function completely and pass "bruno-is-awesome" as the name. My 2 cents.

Reply:
My point of view is how closer to the real scenario, better. So, I think there's no point to change how the name is created here. 

About the property, my bad, sorry. I removed it now.